src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -13,6 +13,40 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
     internal sealed class SourcePropertySymbol : SourcePropertySymbolBase
     {
+        internal bool ContainsFieldKeyword
+        {
+            get
+            {
+                if (CSharpSyntaxNode is not PropertyDeclarationSyntax property)
+                {
+                    return false;
+                }
+
+                if (property.AccessorList is not { } accessorList)
+                {
+                    return false;
+                }
+
+                foreach (var accessor in accessorList.Accessors)
```

Comment:
>accessorList.Accessors [](http://example.com/codeflow?start=41&length=22)

In error scenarios, there could be more accessors in the syntax than we have symbols, I think.

Reply:
@AlekseyTs Yes this is correct. One can have `get => field; get => field;` which will generate "Property accessor already defined" error. But what's the problem with that?

Reply:
>>In error scenarios, there could be more accessors in the syntax than we have symbols, I think.

> Yes this is correct. One can have ```get => field; get => field;`` which will generate "Property accessor already defined" error. But what's the problem with that?

The question is - why look at the accessors that we will never bind, including SemanticModel scenarios? We need to have targeted testing for the error scenarios, including tests for SemanticModel behavior.

Reply:
@AlekseyTs I assume this is no longer relevant after we moved `ContainsFieldKeyword` to the accessor symbol instead of the property symbol?

Reply:
> I assume this is no longer relevant after we moved ContainsFieldKeyword to the accessor symbol instead of the property symbol?

I think we still need to add tests confirming that SemanticModel doesn't bind extra accessors that we ignored. A prototype comment should be good.

Reply:
@AlekseyTs Sorry, I'm not able to figure out the API to use for this test.

```csharp
        [Fact]
        public void ExtraAccessors()
        {
            var comp = CreateCompilation(@"
public class C
{
    public int P { get => field; get => field + 2; }
}
");
            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
            comp.TestOnlyCompilationData = accessorBindingData;

            var model = comp.GetSemanticModel(comp.SyntaxTrees[0]);
            // model.??

            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
        }
```

Reply:
> Sorry, I'm not able to figure out the API to use for this test.

This would require some experimentation.
For example,
- add a real field named "field" (make this a regular property).
- call ```GetSymbolInfo``` on both "field" identifiers
- Now remove the explicit field
- Do the same with ```GetSymbolInfo```
- Compare results between the two property declarations.
- Do we like the behavior?

Reply:
> Now remove the explicit field

Does that mean speculating?

Reply:
> Does that mean speculating?

No. Speculating involves binding code that is not in the original source. Something like, let me see what this expression (new syntax) would mean at this position. Here we are observing behavior for two different declarations, but in both cases, we are asking questions about syntax that is in the original source. To clarify, "Now remove the explicit field" doesn't refer to doing that in context of the same compilation. I mean use two distinct compilations, each with its own form of declaration.

Reply:
@AlekseyTs Ok that's a lot more clear now. Will add all the needed PROTOTYPE comments shortly.

Reply:
@AlekseyTs Actually it looks like `TestSemanticModel_GetterOnly` does part of that (when we don't have explicit field called `field`). So the remaining test is when we have an explicit `field` right?

Reply:
> Actually it looks like `TestSemanticModel_GetterOnly` does part of that (when we don't have explicit field called `field`). So the remaining test is when we have an explicit `field` right?

Does it test behavior within an ignored accessor? The is the most interesting part for this thread.

Reply:
Ah okay. Thanks for the further clarification. Will adjust the prototype